modules/stochastic_tools/src/covariances/SquaredExponentialCovariance.C

```diff
@@ -73,17 +78,18 @@ SquaredExponentialCovariance::SquaredExponentialFunction(RealEigenMatrix & K,
       // Compute distance per parameter, scaled by length factor
       Real r_squared_scaled = 0;
       for (unsigned int kk = 0; kk < num_params_x; ++kk)
-        r_squared_scaled += std::pow((x(ii, kk) - xp(jj, kk)) / length_factor[kk], 2);
-      K(ii, jj) = sigma_f_squared * std::exp(-r_squared_scaled / 2.0);
+        r_squared_scaled +=
+            std::pow((x_accessor[ii][kk] - xp_accessor[jj][kk]) / length_factor[kk], 2);
+      K_accessor[ii][jj] = sigma_f_squared * std::exp(-r_squared_scaled / 2.0);
```

Comment:
Yeah there is an optimized way to do this one, might help. You can use ||x-y||^2 = ||x||^2 + ||y||^2 + 2 x*y.